### ISSUE TITLE
Error building pyspotify when starting via docker-compose

### ISSUE THREAD

**Issue by wuservices** [jaedb/Iris#476]
**Describe the bug**
`pyspotify` fails to build when running `docker-compose up`.

**To Reproduce**
1. Clone repo and cd into folder
2. Set up `docker-compose.yml` and `mopidy.conf`
3. Run `docker-compose-up`

**Expected behavior**
Iris starts. Note that I've never tried to actually run Iris in Docker before, so it's possible I'm just doing something wrong.

**Logs**
```
Building wheels for collected packages: pyspotify
  Building wheel for pyspotify (PEP 517): started
  Building wheel for pyspotify (PEP 517): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python /usr/local/lib/python2.7/dist-packages/pip/_vendor/pep517/_in_process.py build_wheel /tmp/tmpI4kaIj
       cwd: /tmp/pip-install-2XLmE4/pyspotify
  Complete output (60 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-2.7
  creating build/lib.linux-x86_64-2.7/spotify
  copying spotify/search.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/playlist_container.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/playlist_track.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/audio.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/player.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/__init__.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/album.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/utils.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/eventloop.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/sink.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/session.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/social.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/toplist.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/image.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/playlist.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/version.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/_spotify_build.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/error.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/connection.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/compat.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/offline.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/artist.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/playlist_unseen_tracks.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/track.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/link.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/config.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/user.py -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/inbox.py -> build/lib.linux-x86_64-2.7/spotify
  running egg_info
  writing requirements to pyspotify.egg-info/requires.txt
  writing pyspotify.egg-info/PKG-INFO
  writing top-level names to pyspotify.egg-info/top_level.txt
  writing dependency_links to pyspotify.egg-info/dependency_links.txt
  reading manifest file 'pyspotify.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching '*' under directory '.github'
  no previously-included directories found matching 'docs/_build'
  no previously-included directories found matching 'examples/tmp'
  warning: no previously-included files matching '__pycache__/*' found anywhere in distribution
  writing manifest file 'pyspotify.egg-info/SOURCES.txt'
  copying spotify/api.h -> build/lib.linux-x86_64-2.7/spotify
  copying spotify/api.processed.h -> build/lib.linux-x86_64-2.7/spotify
  running build_ext
  generating cffi module 'build/temp.linux-x86_64-2.7/spotify._spotify.c'
  creating build/temp.linux-x86_64-2.7
  building 'spotify._spotify' extension
  creating build/temp.linux-x86_64-2.7/build
  creating build/temp.linux-x86_64-2.7/build/temp.linux-x86_64-2.7
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fdebug-prefix-map=/build/python2.7-2.7.16=. -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/include/python2.7 -c build/temp.linux-x86_64-2.7/spotify._spotify.c -o build/temp.linux-x86_64-2.7/build/temp.linux-x86_64-2.7/spotify._spotify.o
  build/temp.linux-x86_64-2.7/spotify._spotify.c:22:12: fatal error: pyconfig.h: No such file or directory
   #  include <pyconfig.h>
              ^~~~~~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pyspotify
  Running setup.py clean for pyspotify
Failed to build pyspotify
Building wheels for collected packages: Mopidy-Spotify-Web, spotipy, gmusicapi, proboscis, gpsoauth, future, httplib2
  Building wheel for Mopidy-Spotify-Web (setup.py): started
  Building wheel for Mopidy-Spotify-Web (setup.py): finished with status 'done'
  Created wheel for Mopidy-Spotify-Web: filename=Mopidy_Spotify_Web-0.3.0-py2.py3-none-any.whl size=12542 sha256=ee14b803db80baff0dbc262e5d2f112b5b12fbc9862880ec516e381c6309b429
  Stored in directory: /root/.cache/pip/wheels/9a/75/31/d547ab8ef6d1f845d5858869f47b60c793d108a12586bd8168
  Building wheel for spotipy (setup.py): started
  Building wheel for spotipy (setup.py): finished with status 'done'
  Created wheel for spotipy: filename=spotipy-2.4.4-cp27-none-any.whl size=12336 sha256=b4b81232095910ab66ccb507ee293372d4969791d68418c15cb30a9fffdabe69
  Stored in directory: /root/.cache/pip/wheels/76/28/19/a86ca9bb0e32dbd4a4f580870250f5aeef852870578e0427e6
  Building wheel for gmusicapi (setup.py): started
  Building wheel for gmusicapi (setup.py): finished with status 'done'
  Created wheel for gmusicapi: filename=gmusicapi-12.1.1-cp27-none-any.whl size=167219 sha256=e5c9182677009bf7a92b47462672ee44bee4ba76f604bfc9bea546c60aeb543f
  Stored in directory: /root/.cache/pip/wheels/2e/35/80/495bfdd474304bf353f66aa4431e616a45cd1f1fcce02da316
  Building wheel for proboscis (setup.py): started
  Building wheel for proboscis (setup.py): finished with status 'done'
  Created wheel for proboscis: filename=proboscis-1.2.6.0-cp27-none-any.whl size=26439 sha256=30a39bfd92af186ec4afaf4979cb80e3866b5ddf8211de3455bccf9610e3b832
  Stored in directory: /root/.cache/pip/wheels/8c/c9/82/8aa441245194f414659baab25a4f6b3b0d604822c0aff8fb7c
  Building wheel for gpsoauth (setup.py): started
  Building wheel for gpsoauth (setup.py): finished with status 'done'
  Created wheel for gpsoauth: filename=gpsoauth-0.4.1-cp27-none-any.whl size=4802 sha256=1725134bd72799f13607a04aad584b6bea68e016f622e57a3a158c0b1a090598
  Stored in directory: /root/.cache/pip/wheels/a2/14/fc/0754368beb9f2fc1515b7974c23261eac42bd305ff94c6ea4e
  Building wheel for future (setup.py): started
  Building wheel for future (setup.py): finished with status 'done'
  Created wheel for future: filename=future-0.18.2-cp27-none-any.whl size=502568 sha256=208df3842abe0899d73ab5f0469f5e25f30f68cc199b14f69b4781d63ab3bc40
  Stored in directory: /root/.cache/pip/wheels/8b/99/a0/81daf51dcd359a9377b110a8a886b3895921802d2fc1b2397e
  Building wheel for httplib2 (setup.py): started
  Building wheel for httplib2 (setup.py): finished with status 'done'
  Created wheel for httplib2: filename=httplib2-0.15.0-cp27-none-any.whl size=95432 sha256=08889a7d4d927338ed67b0b2f7e6255c7435757ca1eeedaef1425126f75274e4
  Stored in directory: /root/.cache/pip/wheels/f5/f8/3f/87c1cd6981a9b21f41e9224e470b5df72fc6d52aa1dc84e606
Successfully built Mopidy-Spotify-Web spotipy gmusicapi proboscis gpsoauth future httplib2
ERROR: Could not build wheels for pyspotify which use PEP 517 and cannot be installed directly
ERROR: Service 'mopidy' failed to build: The command '/bin/sh -c set -ex  && apt-get update  && DEBIAN_FRONTEND=noninteractive apt-get install -y        curl        dumb-init        gcc        gnupg        gstreamer1.0-alsa        gstreamer1.0-plugins-bad        python-crypto        python-pykka        python-tornado        git        nano        sudo  && curl -L https://apt.mopidy.com/mopidy.gpg | apt-key add -  && curl -L https://apt.mopidy.com/mopidy.list -o /etc/apt/sources.list.d/mopidy.list  && apt-get update  && DEBIAN_FRONTEND=noninteractive apt-get install -y        mopidy        mopidy-soundcloud        mopidy-spotify  && curl -L https://bootstrap.pypa.io/get-pip.py | python -  && pip install -U six pyasn1 requests[security] cryptography  && pip install        Mopidy-Local-Images        Mopidy-Local-SQLite        Mopidy-Spotify-Web        Mopidy-GMusic        Mopidy-Pandora        pyopenssl  && apt-get -f install  && apt-get purge --auto-remove -y        curl        gcc  && apt-get clean  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* ~/.cache  && git clone https://github.com/jaedb/Iris.git /iris  && cd /iris  && python setup.py develop  && mkdir -p /var/lib/mopidy/.config  && ln -s /config /var/lib/mopidy/.config/mopidy  && echo "mopidy ALL=NOPASSWD: /iris/mopidy_iris/system.sh" >> /etc/sudoers' returned a non-zero code: 1
```

**Environment details:**
 - OS: macOS 10.15.2
 - Iris version: latest master

**Additional context**
Saw a similar error in https://github.com/mopidy/mopidy-spotify/issues/249.

**Comment by kingosticks** [comment]
The apt commands install the latest Python 3 compatible versions of Mopidy and Mopidy-spotify, which installs the Python 3 version of pyspotify. But the pip commands are installing Python 2 versions of the extensions. In the case of Mopidy-Spotify-web, it doesn't find a Python 2 version of pyspotify already installed and so tries to build from source, but that doesn't work since it's missing a bunch of build dependencies. In short, this docker file needs to specify a 2.x version of Mopidy to install.

**Comment by wuservices** [comment]
@kingosticks thanks for the insight! I was wondering if this was related to all the Python 3 action, but wasn't sure how to proceed.

By adding a few version numbers in `Dockerfile`, I was able to get it to run:
```
       mopidy=2.\* \
       mopidy-soundcloud=2.\* \
       mopidy-spotify=3.\* \
```

If that looks like the right approach, I can open a PR, but I also wasn't sure if we should just wait until the dust clears, since it looks like https://github.com/jaedb/Iris/issues/356 is moving along.

**Comment by jaedb** [comment]
@wuservices Docker image is now shipped, and Iris `3.44` has been released. This _should_ resolve this issue, but please re-open if the problem persists.
